annotator-core/src/main/java/edu/ucr/cs/riple/core/metadata/trackers/generatedcode/LombokTracker.java

```diff
@@ -0,0 +1,41 @@
+package edu.ucr.cs.riple.core.metadata.trackers.generatedcode;
+
+import edu.ucr.cs.riple.core.metadata.method.MethodDeclarationTree;
+import edu.ucr.cs.riple.core.metadata.trackers.MethodRegionTracker;
+import edu.ucr.cs.riple.core.metadata.trackers.Region;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class LombokTracker implements GeneratedRegionTracker {
+
+  private final MethodRegionTracker tracker;
+  private final MethodDeclarationTree methodDeclarationTree;
+
+  public LombokTracker(
+      MethodDeclarationTree methodDeclarationTree, MethodRegionTracker methodRegionTracker) {
+    this.methodDeclarationTree = methodDeclarationTree;
+    this.tracker = methodRegionTracker;
+  }
+
+  @Override
+  public Set<Region> extendWithGeneratedRegions(Set<Region> regions) {
+    return regions.stream()
+        // filter regions where are created by lombok
```

Comment:
I think the idea here is that for any Lombok-generated method in `regions`, we also add all _uses_ of that method to the impacted regions.  Is this the right idea?  If so, let's add some comments / Javadoc documenting the logic here.  Also we should state briefly why we think this is sufficient to get all the relevant regions, given the documented assumptions on how Lombok propagates annotations (see previous comment)

Reply:
f0880e2067988c53d64832202e269c34e7366588